### PR TITLE
fix #770 date keyword in Python >= 3.7

### LINKED ISSUE #770: The "date" keyword is ignored by the from_json parser
The BIOME format has a keyword "date" in the top level. The `Table` has an attribute `create_date` which is `None` after loading a `.biom` file though. Could this field be parsed into the `Table`?

To reproduce:
```python
from json import loads
from pathlib import Path
import numpy as np
from biom import load_table
from biom.table import Table
data = np.arange(40).reshape(10, 4)
sample_ids = ['S%d' % i for i in range(4)]
observ_ids = ['O%d' % i for i in range(10)]
sample_metadata = [{'environment': 'A'}, {'environment': 'B'},
                   {'environment': 'A'}, {'environment': 'B'}]
observ_metadata = [{'taxonomy': ['Bacteria', 'Firmicutes']},
                   {'taxonomy': ['Bacteria', 'Firmicutes']},
                   {'taxonomy': ['Bacteria', 'Proteobacteria']},
                   {'taxonomy': ['Bacteria', 'Proteobacteria']},
                   {'taxonomy': ['Bacteria', 'Proteobacteria']},
                   {'taxonomy': ['Bacteria', 'Bacteroidetes']},
                   {'taxonomy': ['Bacteria', 'Bacteroidetes']},
                   {'taxonomy': ['Bacteria', 'Firmicutes']},
                   {'taxonomy': ['Bacteria', 'Firmicutes']},
                   {'taxonomy': ['Bacteria', 'Firmicutes']}]
table = Table(data, observ_ids, sample_ids, observ_metadata,
              sample_metadata, table_id='Example Table')
file = Path('~/test.biom').expanduser()
table.to_json('me', file.open('w'))
loaded = load_table(file.as_posix())
print(loaded.create_date)  # is None
loaded_2 = Table.from_json(loads(table.to_json('me')))
print(loaded_2.create_date)  # is None
```

This is quite similar to #697. (I don't know why it was closed...)
Edit: It was closed because of commit b1fff87f9ca3a9f3c19313c98436e2e21535b33e.

### PR DESCRIPTION
**Pull request by schferbe**
This is a revamp of #771 without adding external dependencies.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 BIOM-Format ChangeLog
 =====================
 
+biom 2.1.11
```
Comment (wasade) [inline ChangeLog.md:4]:
```suggestion
biom 2.1.11-dev
```

**Review by wasade — CHANGES_REQUESTED** [review]
Just some minor items, thanks!!

**Comment by schferbe** [comment]
I applied your suggestions. Thanks for the review!

**Comment by wasade** [comment]
Thanks, @schferbe!

**Pull request merged**